tools/src/verdaccio/verdaccio.plugin.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  type CreateNodes,
+  type CreateNodesContext,
+  readJsonFile,
+} from '@nx/devkit';
+import { dirname } from 'node:path';
+import { type ProjectConfiguration } from 'nx/src/config/workspace-json-project-json';
+import { uniquePort } from './utils';
+
+type CreateNodesOptions = {
+  // @TODO move into target options
+  port?: string | number;
+  config?: string;
+  storage?: string;
+  preTargets?: string | string[];
+  verbose?: boolean;
+};
+
+export const createNodes: CreateNodes = [
+  '**/project.json',
+  (
+    projectConfigurationFile: string,
+    opts: undefined | unknown,
+    context: CreateNodesContext,
+  ) => {
+    const {
+      port = uniquePort(),
+      config = '.verdaccio/config.yml',
+      storage = 'tmp/local-registry/storage',
+      verbose = false,
+      preTargets = ['e2e'],
+    } = (opts ?? {}) as CreateNodesOptions;
+    const { workspaceRoot } = context;
+    const root = dirname(projectConfigurationFile);
+    const projectConfiguration: ProjectConfiguration = readJsonFile(
+      projectConfigurationFile,
+    );
+
+    // @TODO in the future the root project has no verdaccio target anymore. only e2e projects have.
+    const isRootProject = root === '.';
+    if (!isRootProject) {
+      return {};
+    }
+
+    return {
+      projects: {
+        [root]: {
+          targets: verdaccioTargets({
+            port,
+            config,
+            storage,
+            verbose,
+            preTargets,
+          }),
+        },
+      },
+    };
+  },
+];
+
+function verdaccioTargets({
+  port,
+  config,
+  storage,
+  preTargets,
+}: Required<CreateNodesOptions>) {
+  const targets = Array.isArray(preTargets) ? preTargets : [preTargets];
+  return {
+    'start-verdaccio': {
+      executor: '@nx/js:verdaccio',
+      options: {
+        port,
+        config,
+        storage,
+      },
+    },
+  };
```

Comment:
this target originally lived in [`project.json`](https://github.com/code-pushup/cli/pull/798/files#diff-7680eb66f8e7b2155dfc25c4aa16a606a8fd3decb988265f218f3e11a8afa2be)